test/memory/lifetime/allocation.chpl

```diff
@@ -0,0 +1,74 @@
+use Time;
+
+config const numTrials : int = 10;
```

Comment:
You probably know this, but just FYI that running multiple trials and not timing the first one likely avoids timing the overhead of mmap. It's very likely jemalloc won't have returned the memory back to the system for the second trial, so it'll re-use memory it got from the system.

That's likely what you want in this case since I think the intention is to measure Chapel overheads. Might be worth adding a quick comment at the top of this file to describe the intent.

Reply:
I do skip the first trial. Is the suggestion that I add documentation describing _why_ I skip the first trial? Normally my explanation for doing so is "warming up the cache". (Note that in the loop over trials, I deliberately do `0..numTrials` and skip when `trials == 0`.)

Reply:
In this case it's more than just a warmup trial since the first trial is also what will result in grabbing memory from the system. 

I don't think you need a big comment, just something short that describes what this test is intending to measure (which I think in this case is Chapel overheads for memory allocation styles, and not so much the underlying allocator time.)